From d4d97d30194e82266af828060350f8c07aa4e4a8 Mon Sep 17 00:00:00 2001
From: Alfred Chen <cchalpha@gmail.com>
Date: Wed, 8 Jul 2020 10:58:57 +0800
Subject: [PATCH 003/148] sched/bmq: Adjust MAX_PRIORITY_ADJ to 7

Adjust task boost priority to +/-7, and change SCHED_BITS to 128.
---
 include/linux/sched/prio.h | 2 +-
 kernel/sched/alt_core.c    | 4 ++--
 kernel/sched/bmq.h         | 4 ++--
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/include/linux/sched/prio.h b/include/linux/sched/prio.h
index c28676e431be..dc7c5c159749 100644
--- a/include/linux/sched/prio.h
+++ b/include/linux/sched/prio.h
@@ -20,7 +20,7 @@
 
 #ifdef CONFIG_SCHED_ALT
 /* +/- priority levels from the base priority */
-#define MAX_PRIORITY_ADJ	4
+#define MAX_PRIORITY_ADJ	7
 #endif
 
 /*
diff --git a/kernel/sched/alt_core.c b/kernel/sched/alt_core.c
index 09ca47de425c..7316079ef171 100644
--- a/kernel/sched/alt_core.c
+++ b/kernel/sched/alt_core.c
@@ -78,7 +78,7 @@ int sched_yield_type __read_mostly = 1;
 
 #define rq_switch_time(rq)	((rq)->clock - (rq)->last_ts_switch)
 #define boost_threshold(p)	(sched_timeslice_ns >>\
-				 (10 - MAX_PRIORITY_ADJ -  (p)->boost_prio))
+				 (15 - MAX_PRIORITY_ADJ -  (p)->boost_prio))
 
 static inline void boost_task(struct task_struct *p)
 {
@@ -185,7 +185,7 @@ static inline void update_sched_rq_watermark(struct rq *rq)
 
 static inline int task_sched_prio(struct task_struct *p)
 {
-	return (p->prio < MAX_RT_PRIO)? p->prio : p->prio + p->boost_prio;
+	return (p->prio < MAX_RT_PRIO)? p->prio : MAX_RT_PRIO / 2 + (p->prio + p->boost_prio) / 2;
 }
 
 #include "bmq_imp.h"
diff --git a/kernel/sched/bmq.h b/kernel/sched/bmq.h
index 4ce30c30bd3e..aba3c98759f8 100644
--- a/kernel/sched/bmq.h
+++ b/kernel/sched/bmq.h
@@ -2,8 +2,8 @@
 #define BMQ_H
 
 /* bits:
- * RT(0-99), Low prio adj range, nice width, high prio adj range, cpu idle task */
-#define SCHED_BITS	(MAX_RT_PRIO + NICE_WIDTH + 2 * MAX_PRIORITY_ADJ + 1)
+ * RT(0-99), (Low prio adj range, nice width, high prio adj range) / 2, cpu idle task */
+#define SCHED_BITS	(MAX_RT_PRIO + NICE_WIDTH / 2 + MAX_PRIORITY_ADJ + 1)
 #define IDLE_TASK_SCHED_PRIO	(SCHED_BITS - 1)
 
 struct bmq {
-- 
2.32.0.93.g670b81a890

